mds/management/commands/poll_providers.py

```diff
@@ -199,15 +200,30 @@ def get_body(self, provider, url):
     def process_status_changes(self, status_changes, provider):
         if self.verbosity > 1:
             self.stdout.write("Processing...")
+
+        # What is the latest event recorded in that series? (order is not asserted in the specs)
+        try:
+            last_start_time_polled = utils.from_mds_timestamp(
+                max(status_change["event_time"] for status_change in status_changes)
+            )
+        except (KeyError, ValueError):
+            # Data so bad there is no or nothing but invalid event times
+            logger.exception("No valid event_time found in status_changes series: %s", status_changes)
+            # How can we prevent from asking them again next time?
+            if provider.last_start_time_polled:
+                return provider.last_start_time_polled + datetime.timedelta(milliseconds=1)
```

Comment:
I don't know this code. As I understand it, `provider.last_start_time_polled` was set at "Now - 1hour" when you polled one hour ago with an event that occured 1 hour ago. Here you want to process the newer events and set a new `provider.last_start_time_polled`. But you may not find a correct timestamp in events. What is the benefit of adding a millisecond ? It will avoid polling the whole device timeline from the beginning but with super low increments no ? So that you're still polling data very far in the past

Reply:
I'm slowly moving forward the the time window until I find the first valid data. As `event_time` is in milliseconds, 1 millisecond is the smallest unit.

If I was moving to the next second, I could potentially skip a thousand events.